src/ussd_popi_user_data.js

```diff
@@ -316,16 +322,35 @@ go.app = function() {
                 ],
                 next: function(choice) {
                     return self.im.user
-                    // TODO: update on IdentityStore
                     .set_lang(choice.value)
                     .then(function() {
-                        self.im.user.set_answer("state_language", choice.value);
-                        return 'state_updated';
+                        self.im.user.set_answer("new_language", choice.value);
+                        return 'state_switch_lang';
                     });
                 },
             });
         });
 
+        self.add('state_switch_lang', function(name) {
+            var change_info = {
+                "registrant_id": self.im.user.answers.operator.id,
+                "action": "momconnect_change_language",
+                "data": {
+                    "language": self.im.user.answers.new_language,
+                    "old_language": self.im.user.answers.operator.details.lang_code
```

Comment:
I've created a pull request for this here: https://github.com/praekeltfoundation/ndoh-hub/pull/101